src/lib/utils.rs

```diff
@@ -71,12 +72,17 @@ pub fn filenames<P: AsRef<Path>>(
 ) -> Vec<PathBuf> {
     let mut output_paths = vec![];
     let mut counter = AHashMap::new();
+    let output_dir = match &sample.project {
+        Some(project) => output_dir.as_ref().join(project.to_string()),
+        None => output_dir.as_ref().to_path_buf(),
+    };
+    std::fs::create_dir_all(&output_dir).unwrap();
```

Comment:
I think instead you could do this at `run.rs:93` where it would feel more at home and allow for better error propogation if the directory can't be created.